src/elements/content-sidebar/activity-feed/comment/components/BaseCommentMenuWrapper.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import React from 'react';
+import type { BoxCommentPermission, FeedItemStatus } from '../../../../../common/types/feed';
+import type { OnAnnotationEdit, OnCommentEdit } from '../types';
+
+import { BaseCommentMenu } from './BaseCommentMenu';
+
+export interface BaseCommentMenuWrapperProps {
+    canDelete: boolean;
+    canEdit: boolean;
+    canResolve: boolean;
+    id: string;
+    isEditing: boolean;
+    isInputOpen: boolean;
+    isResolved: boolean;
+    onAnnotationEdit?: OnAnnotationEdit | typeof undefined;
+    onCommentEdit: OnCommentEdit;
+    onDelete: ({ id: string, permissions?: BoxCommentPermission }) => any;
+    onSelect: (isSelected: boolean) => void;
+    permissions: BoxCommentPermission;
+    setIsEditing: ((boolean => boolean) | boolean) => void;
+    setIsInputOpen: ((boolean => boolean) | boolean) => void;
+}
+
+export const BaseCommentMenuWrapper = ({
+    canDelete,
+    canEdit,
+    canResolve,
+    id,
+    isEditing,
+    isInputOpen,
+    isResolved,
+    onAnnotationEdit,
+    onCommentEdit,
```

Comment:
Should these be combined into `onEdit`? We can allow the parent to figure out comment versus annotation.

Reply:
That's part of why this wrapper exists. This way, the dumb component `BaseCommentMenu` doesn't need to know about `onCommentEdit` and `onAnnotationEdit`. It just receives `handleStatusUpdate`.